analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GenericTypeParameterUnused.cs

```diff
@@ -180,11 +180,41 @@ public void Method(Tests.Diagnostics.T x)
         }
     }
 
-    public struct SomeStruct<T> // FN
+    public struct SomeStruct<T> // Noncompliant
     {
         public int Add<T>(int a, int b) // Noncompliant; <T> is ignored
         {
             return a + b;
         }
     }
+
+    public struct StructUnused<T>   // Noncompliant {{'T' is not used in the struct.}}
+    {
+    }
+
+    public interface IUsedAsReturnType<T>
+    {
+        T Create();
+    }
+
+    public interface IUsedAsProperty<T>
+    {
+        T Value { get; set; }
+    }
+
+    public interface IUsedAsArgument<T>
+    {
+        object Create(T arg);
+    }
+
+    public interface IUnusedEmpty<T>    // Noncompliant {{'T' is not used in the interface.}}
+    {
+    }
+
```

Comment:
```suggestion
public interface IUnusedWithVariance<out T>
{
}

public interface IUsedInBaseType<T>: IEnumerable<T>
{
}

public interface IUsedInTypeConstraint<T> where T: class
{
}
```

Reply:
As discussed offline, I've added them as noncompliant. They are not practically used by the interface.